include/bluetooth/a2dp.h

```diff
@@ -34,6 +34,23 @@ static struct bt_a2dp_endpoint bt_a2dp_endpoint_##name = \
 };\
 static struct bt_a2dp_endpoint *name = &bt_a2dp_endpoint_##name
 
+/** @brief define the default SBC sink endpoint that can be used as bt_a2dp_register_endpoint's parameter.
+ *
+ * SBC is mandatory as a2dp specification, A2DP_SBC_DEFAULT_SINK_ENDPOINT is more convenient for user to register SBC endpoint.
+ *
+ *  @param name the endpoint variable name.
+ */
+#define A2DP_SBC_DEFAULT_SINK_ENDPOINT(name)\
+static uint8_t bt_a2dp_endpoint_cap_buffer##name[5] = {4U, A2DP_SBC_SAMP_FREQ_44100 | A2DP_SBC_SAMP_FREQ_48000 | A2DP_SBC_CH_MODE_MONO | A2DP_SBC_CH_MODE_STREO | A2DP_SBC_CH_MODE_JOINT, A2DP_SBC_BLK_LEN_16 | A2DP_SBC_SUBBAND_8 | A2DP_SBC_ALLOC_MTHD_LOUDNESS, 18U, 35U};\
+static struct bt_a2dp_endpoint bt_a2dp_endpoint_##name = \
+{\
+	.codec_id = BT_A2DP_SBC,\
+	.info = {.sep = {.media_type = BT_A2DP_AUDIO, .tsep = BT_A2DP_SOURCE}, .next = NULL},\
+	.preset = NULL,\
+	.caps = (struct bt_a2dp_preset *)&bt_a2dp_endpoint_cap_buffer##name[0],\
+};\
+static struct bt_a2dp_endpoint *name = &bt_a2dp_endpoint_##name
+
```

Comment:
This should probably be made generic so other codecs can be initialized with the macro e.g:

#define BT_A2DP_ENDPOINT(name, type, codec, caps) \
{
...(fill bt_a2dp_endpoint)
}
#define BT_A2DP_SINK(name, codec, caps) BT_A2DP_ENDPOINT(name, BT_A2DP_SINK, codec, caps)
#define BT_A2DP_SOURCE(name, codec, caps) BT_A2DP_ENDPOINT(name, BT_A2DP_SOURCE, codec, caps)

The for have SBC specific macros:
#define BT_A2DP_SBC_SOURCE(name, caps) BT_A2DP_SOURCE(name, BT_A2DP_SBC, caps)
#define BT_A2DP_SBC_SINK(name, caps) BT_A2DP_SINK(name, BT_A2DP_SBC, caps)

Note if we want to really declare it statically we could use Z_STRUCT_SECTION_ITERABLE to build it as part of the rom, but that can be done later on if we find it really worth adding another section in the binary.

Reply:
Has updated the codes, please check it again Thanks.
I think we don't need to put it in section. The usage is as follow:
case1: sbc source endpoint.
BT_A2DP_SBC_SOURCE_ENDPOINT(sbcEndpoint, A2DP_SBC_SAMP_FREQ_44100);
bt_a2dp_register_endpoint(&sbcEndpoint, BT_A2DP_AUDIO, BT_A2DP_SOURCE);
case2: sbc sink endpoint.
BT_A2DP_SBC_SINK_ENDPOINT(sbcEndpoint);
bt_a2dp_register_endpoint(&sbcEndpoint, BT_A2DP_AUDIO, BT_A2DP_SINK);